mmv1/templates/terraform/pre_create/privateca_certificate_authority.go.erb

```diff
@@ -0,0 +1,8 @@
+if p, ok := d.GetOk("desired_state"); ok {
```

Comment:
The logic for determining a valid state change should be done in a customize diff function, not in a pre-create and pre-update. This will allow users to get invalid transition error during plan-time instead of apply-time.

In a customize diff, you can get the old and new version of a field with `diff.GetChange`. See an example here: https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/templates/terraform/constants/spanner_database.go.erb

You need two lines in the `terraform.yaml`, a constant and resource_definition. See https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/products/spanner/terraform.yaml#L69
and 
https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/templates/terraform/resource_definition/spanner_database.go.erb

Reply:
The logic here is only for CA creation.

How do we check if `CustomizeDiff` is called for resource creation vs resource update?

Reply:
You can check the "before" value of a field that would otherwise always have a value, like `name` in this example: https://github.com/GoogleCloudPlatform/magic-modules/blob/a2ab8e4ebe1e2a53160a43d5761a53d846ffef31/mmv1/third_party/terraform/resources/resource_bigtable_instance.go#L464

Reply:
Thanks! Have updated the PR as suggested.